lua-mosquitto.c

```diff
@@ -432,6 +432,56 @@ static int ctx_tls_set(lua_State *L)
 	return mosq__pstatus(L, rc);
 }
 
+struct mosq_opt_string {
+	const char *str;
+	enum mosq_opt_t tls_opt;
+};
+
+/***
+ * Enables the use of a TLS engine device.
+ * @function string_option
+ * @tparam[opt=nil] string option may be nil
+ * @tparam[opt=nil] string value may be nil
+ * @see mosquitto_string_option
+ * @return[1] boolean true
+ * @return[2] nil
+ * @treturn[2] number error code
+ * @treturn[2] string error description.
+ * @raise For some out of memory or illegal states
+ */
+static int ctx_string_option(lua_State *L)
+{
+       ctx_t *ctx = ctx_check(L, 1);
+       const char *mosq_option = luaL_optstring(L, 2, NULL);
```

Comment:
you're just going to reject it anyway if it's null, why not use a normal check here?